File: solutions/java/lasagna/1/src/main/java/Lasagna.java
public class Lasagna {
    
    public int expectedMinutesInOven() {
        return 40;  
    }

    public int remainingMinutesInOven(int actualMinutesInOven) {
        return expectedMinutesInOven() - actualMinutesInOven;
    }
    
    public int preparationTimeInMinutes(int layers) {
       
        int minutesPerLayer = 2;

        return layers * minutesPerLayer;
    }

    public int totalTimeInMinutes(int layers, int actualMinutesInOven) {
        return preparationTimeInMinutes(layers) + actualMinutesInOven;
    }
    
}

File: solutions/ruby/lasagna/1/lasagna.rb
class Lasagna
  EXPECTED_MINUTES_IN_OVEN = 40;
  MINUTES_PER_LAYER = 2;
  def remaining_minutes_in_oven(actual_minutes_in_oven)
    EXPECTED_MINUTES_IN_OVEN - actual_minutes_in_oven
  end

  def preparation_time_in_minutes(layers)
    MINUTES_PER_LAYER*layers
  end

  def total_time_in_minutes(number_of_layers:, actual_minutes_in_oven:)
    return actual_minutes_in_oven + (number_of_layers*MINUTES_PER_LAYER)
  end
end

lasagna = Lasagna.new
lasagna.remaining_minutes_in_oven(30)
lasagna = Lasagna.new
lasagna.preparation_time_in_minutes(2)
lasagna = Lasagna.new
lasagna.total_time_in_minutes(number_of_layers: 3, actual_minutes_in_oven: 20)
